### PR TITLE
Update flat_field reffile docs

### PR DESCRIPTION
**Pull request by hbushouse**
More updates for #2675, this time for the `flat_field` step.

### CODE REVIEW

**Comment by hbushouse** [comment]
@philhodge Mainly just cosmetic and minor wording changes here and there, with the addition of some new info, to try to get it closer to matching the organization and content of other step docs (although the NIRSpec modes are so complicated that it doesn't match exactly).

One major question: The existing section for the NIRSpec FFLAT in IFU mode contained the statement "The flat field values in this table are used to account for a wavelength-dependence on a much finer scale than given by the values in the SCI array." But the IFU form of the FFLAT doesn't contain a SCI extension (only FAST_VARIATION). So was that a copy-n-paste error from other sections and hence should be removed?

File: docs/jwst/flatfield/reference_files.rst [hunk 1]
```diff
@@ -295,10 +331,7 @@ The FAST_VARIATION table contains four columns:
...
 
-
-The flat field values in this table are used to account for a wavelength-dependence on a much
-finer scale than given by the values in the SCI array.  There is a single row in this table, 
-as the same wavelength-dependent value is applied to all pixels.
-
-.. include:: ../includes/dq_def.rst
+The flat field values in this table are used to account for a wavelength-dependence on a
+much finer scale than given by the values in the SCI array.  There is a single row in
+this table, because the same wavelength-dependent value is applied to all pixels.
```
Comment (philhodge) [inline docs/jwst/flatfield/reference_files.rst]: No, the wavelength and data in that row are arrays.

File: docs/jwst/flatfield/reference_files.rst [hunk 2]
```diff
@@ -78,45 +101,49 @@ there are different flat fields for fixed-slit data, IFU data, and for
...
-in the "slit_name" column.  For multi-object spectroscopic data, the flat
-field contains four sets (one for each MSA quadrant) of images, WAVELENGTH
-tables, and FAST_VARIATION tables.  The images are unique to the fore
-optics flat fields, however.  The image "pixels" correspond to micro-shutter
-array slits, rather than to detector pixels.  The array size is 365 rows
+in the "slit_name" column.  For multi-object spectroscopic data, the FFLAT
+contains four sets of images (one for each MSA quadrant), WAVELENGTH
+tables, and FAST_VARIATION tables.  The images are unique to the FFLATs,
+however.  The image "pixels" correspond to micro-shutter array slits, rather
+than to detector pixels.  The array size is 365 rows
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply docs/jwst/flatfield/reference_files.rst]: Yes, it looks like it is 365 columns (x-axis) by 171 rows (y-axis). That was part of the original text. I'll update.

File: docs/jwst/flatfield/reference_files.rst [hunk 3]
```diff
@@ -78,45 +101,49 @@ there are different flat fields for fixed-slit data, IFU data, and for
...
 
+FFLAT Reference File
+--------------------
+
+:REFTYPE: FFLAT
 
-Fore Optics (FFLAT)
-:::::::::::::::::::
-There are 3 types of FFLAT reference files: fixed slit, msa spec, and IFU. For each type
+There are 3 forms of FFLAT reference files: fixed slit, MSA spec, and IFU. For each type
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply docs/jwst/flatfield/reference_files.rst]: Probably don't need to say it. It's a standard statement that we include for all FITS ref files. I do think the term "primary" is appropriate though. In FITS lingo "primary" usually refers to the first, regular (non-extension) HDU, i.e. the one that's formatted according to the original FITS specs. The term "primary" was adopted for it when extensions were invented. I don't think it refers to the first HDU that actually has data in it. Regarding the use of "empty", I guess you're right that it's really more than just empty - it's not there at all. The only part of the primary HDU that exists is the header. I guess we could rephrase it to something like "the primary HDU does not contain a data array" (so it's just an "HU" instead of an "HDU"!).

File: docs/jwst/flatfield/reference_files.rst [hunk 4]
```diff
@@ -126,56 +153,61 @@ The table contains four columns:
...
+FAST_VARIATION BINTABLE   2   TFIELDS = 4           N/A
+DQ_DEF         BINTABLE   2   TFIELDS = 4           N/A
+============== ======== ===== ===================== =========
 
+.. include:: ../includes/dq_def.rst
 
+For the 5 extensions that appear multiple times, the EXTVER keyword indicates the
+quadrant number, 1 to 4.
+Each plane of the SCI array gives the flat_field value for every pixel in the quadrant for
+the corresponding wavelength, which is specified in the WAVELENGTH table.
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply docs/jwst/flatfield/reference_files.rst]: OK, I'll incorporate that into an update.

File: docs/jwst/flatfield/reference_files.rst [hunk 5]
```diff
@@ -126,56 +153,61 @@ The table contains four columns:
...
-
-The flat field values in this table are used to account for a wavelength-dependence on a much
-finer scale than given by the values in the SCI array.  There is a single row in this table, 
-as the same wavelength-dependent value is applied to all pixels in the quadrant.
-
-.. include:: ../includes/dq_def.rst
+The flat field values in this table are used to account for a wavelength-dependence on a
+much finer scale than given by the values in the SCI array.  There is a single row in
+this table, as the same wavelength-dependent value is applied to all pixels in the
+quadrant.
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply docs/jwst/flatfield/reference_files.rst]: Updated in new version.

File: docs/jwst/flatfield/reference_files.rst [hunk 6]
```diff
@@ -212,75 +245,78 @@ extension labeled FAST_VARIATION. The table contains four columns:
...
 * data: float 1-D array, flat field values for each wavelength
 
-The flat field values in this table are used to account for a wavelength-dependence on a much
-finer scale than given by the values in the SCI array.  For each pixel in the science data, 
-the wavelength of the light that fell on that pixel will be determined by using the WCS
-interface.  The flat-field value for that pixel will then be obtained by
+The flat field values in this table are used to account for a wavelength-dependence on a
+much finer scale than given by the values in the SCI array.  For each pixel in the
+science data, the wavelength of the light that fell on that pixel will be determined by
+using the WCS interface.  The flat-field value for that pixel will then be obtained by
```
[the remark this answers is not in the document]
Comment (hbushouse) [reply docs/jwst/flatfield/reference_files.rst]: Updated in new version.

File: docs/jwst/flatfield/reference_files.rst (hunk 1 above)
Comment (hbushouse) [reply docs/jwst/flatfield/reference_files.rst]: fixed in new version
